### ISSUE TITLE
Messages: Filters: Search filter does not work for value equal to null

### ISSUE THREAD

**Issue by BaoKhanhFontAriel** [kafbat/kafka-ui#630]
### Issue submitter TODO list

- [X] I've looked up my issue in [FAQ](https://ui.docs.kafbat.io/faq/common-problems)
- [X] I've searched for an already existing issues [here](https://github.com/kafbat/kafka-ui/issues)
- [X] I've tried running `main`-labeled docker image and the issue still persists there
- [X] I'm running a supported version of the application which is listed [here](https://github.com/kafbat/kafka-ui/blob/main/.github/SECURITY.md)

### Describe the bug (actual behavior)

I add filter record.value.name == null to search all values that have name equal to null but it shows no message found

### Expected behavior

There should be 1 message found with name == null

### Your installation details

Version is kafbat-ui-v1.0.0.jar, I run it with jar file

### Steps to reproduce

1. Produce 2 message {"name":"Sam", "age": 25} and {"name": null, "age": 24}
2. Add filter record.value.name == null
3. Result shows no message found

### Screenshots

![image](https://github.com/user-attachments/assets/376c6a7d-6cc2-4e35-ac39-1dc01b31c247)
![image](https://github.com/user-attachments/assets/c15c8268-08a6-494d-a0c7-6b4b90f9c94b)
![image](https://github.com/user-attachments/assets/9f382568-6c28-49f9-887c-1e94c69f4506)

**Comment by sixdouglas** [comment]
Hi, I'd be happy to work on this issue

**Comment by sixdouglas** [comment]
I've played a little around this issue. I can reproduce the behaviour in a unit simple test, using just the plain CEL library. 

I found this in the CEL lib documentation ( [here](https://github.com/google/cel-spec/blob/master/doc/langdef.md#booleans-and-null) ):

> The null value is written null. It is used in conversion to and from protocol buffer and JSON data, but otherwise has no built-in meaning in CEL. In particular, null has its own type (null_type) and is not necessarily allowed where a value of some other type is expected.

I didn't manage to compare anything (the `name` key in the example below) to `null`. Using an expression like `record.value.name == null` to validate on a map like this : 

```
Map<String, Object> objectKeys = new HashMap<>();
objectKeys.put("name", null);
objectKeys.put("age", 24);
Map<String, Object> valueKeys = new HashMap<>();
valueKeys.put("value", objectKeys);
Map<String, Map<String, Object>> recordKeys = new HashMap<>();
recordKeys.put("record", valueKeys);
```

Gives me an output result like this: 

```
unknown {
  exprs: 3
}
```

Do you know a specific way to do this kind of comparison using this library?

**Comment by yeikel** [comment]
Can you send a draft PR with your progress so far? I'd like to try/experiment what the change looks like along with you

**Comment by Haarolean** [comment]
> I didn't manage to compare anything 

yeah me neither, I've tried `typeof` and some other things, nothing worked (which worked fine here: https://playcel.undistro.io/ ). Most likely, our CEL env requires adjustments (see `MessageFilters#createCompiler` and `MessageFilters#createRuntime`).

**Comment by sixdouglas** [comment]
Hi,
Here is my PR.
It is just a draft, since it doesn't work ;)
I've added multiple tests, one using the CEL as it is setup in the app. And two others completely outside the app, to see if there was something special in the setup.
Testing String and Number equality and builtin functions works but not number inequality nor null checks

**Comment by Haarolean** [comment]
@yeikel wanna take a look at this?

**Comment by yeikel** [comment]
> @yeikel wanna take a look at this?

Yes. Please assign it to me

**Comment by helpermethod** [comment]
This is a huge blocker for us, as we often have to filter for null values.

**Comment by Haarolean** [comment]
#1050 fixes it (thanks @germanosin) for json-based messages, avro is trickier -- it annihilates all null fields and the (de-)serialization here is done via avro library. I'm gonna stop here with json schemas, if someone needs avro, we can reopen and retake a look.
